### PR TITLE
test: Adding premium disk support to ss and deployments on e2e tests.

### PR DESCRIPTION
**Pull request by landreasyan**
…Also adding an extender e2e test with volume attachments replicas.

**What type of PR is this?**
Adds support of premium disks with maxshares>1 to statefulsets and deployments in e2e test suite. Also adds a scheduler extender test with volumes with maxshare=2 to check for scheduling in the presence of replica volume attachments.  

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by landreasyan** [comment]
Replica azvolumeattachments are currently not being created for the maxshare>0 case. I am investigating the issue and plan to merge this PR once I have a root cause.

**Review by edreed — APPROVED** [review]

**Comment by edreed** [comment]
It looks like the e2e test failures are fixed by #88. It would be good to rebase on master to get a clean test run.

File: pkg/azdiskschedulerextender/azdiskschedulerextender_metrics.go [hunk 1]
```diff
@@ -51,6 +51,20 @@ var (
...
 		[]string{"filter"})
 
+	SchedulerExtenderRequestCount = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Namespace: scheduler,
+			Name:      "extender_requests_total",
+			Help:      "Total extender request.",
+		})
+
+	// PrioritizeNodesRequestCount = metrics.NewCounter(
```
Comment (landreasyan) [inline pkg/azdiskschedulerextender/azdiskschedulerextender_metrics.go:61]: Will be removing `PrioritizeNodesRequestCount` and `CacheSize` as those are not used for the scheduler extender.

File: charts/index.yaml [hunk 2]
```diff
@@ -3,119 +3,119 @@ entries:
...
     appVersion: v2.0.0-alpha.1
-    created: "2021-07-16T18:41:00.670264Z"
+    created: "2021-07-28T11:38:06.413480342-07:00"
     description: Azure disk Container Storage Interface (CSI) Storage Plugin
-    digest: dde8753005b06810bce75930d7d4b0be634e962d0cfb7f4e8288a5112b7fa80d
+    digest: d784455a2f3784d989ef75b48f7f1a99773a89e36a513257b122c89590b42e4f
     name: azuredisk-csi-driver
     urls:
-    - https://raw.githubusercontent.com/abhisheksinghbaghel/azuredisk-csi-driver/master/charts/v2.0.0-alpha.1/azuredisk-csi-driver-v2.0.0-alpha.1.tgz
+    - https://raw.githubusercontent.com/abhisheksinghbaghel/azuredisk-csi-driver/master/chart/v2.0.0-alpha.1/azuredisk-csi-driver-v2.0.0-alpha.1.tgz
```
Comment (edreed) [inline charts/index.yaml:11]: It looks like the wrong URL was used when the repo index was updated.

**Review by edreed — CHANGES_REQUESTED** [review]
Need to fix the URL in the Helm chart repo index, but otherwise LGTM.

**Review by edreed — APPROVED** [review]

**Comment by landreasyan** [comment]
The failure in integration test is fixed by @mukhoakash's PR - https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/4580be6335a126a69bc1fdcdb53beb183296356e/pkg/provisioner/crdprovisioner.go#L562

**Pull request merged**